webroot/js/app.js

```diff
@@ -192,141 +162,319 @@ class Owncast {
-  };
 
+  handleUsernameChange(newName, newAvatar) {
+    this.setState({
+      username: newName,
+      userAvatarImage: newAvatar,
+    });
+  }
 
-  handlePlayerPlaying() {
-    // do something?
-  };
+  handleChatPanelToggle() {
+    const { displayChat: curDisplayed } = this.state;
+
+    const displayChat = !curDisplayed;
+    if (displayChat) {
+      setLocalStorage(KEY_CHAT_DISPLAYED, displayChat);
+		} else {
+			clearLocalStorage(KEY_CHAT_DISPLAYED);
+		}
+    this.setState({
+      displayChat,
+    });
+  }
 
+  disableChatInput() {
+    this.setState({
+      chatEnabled: false,
+    });
+  }
 
-  // likely called some time after stream status has gone offline.
-  // basically hide video and show underlying "poster"
-  handlePlayerEnded() {
-    this.vueApp.playerOn = false;
-  };
+  handleNetworkingError(error) {
+    console.log(`>>> App Error: ${error}`);
+  }
 
-  handlePlayerError() {
-    // do something?
-    this.handleOfflineMode();
-    this.handlePlayerEnded();
-  };
-};
+  handleWindowResize() {
+    this.setState({
+      windowWidth: window.innerWidth,
+      windowHeight: window.innerHeight,
+    });
+  }
+
+  render(props, state) {
+    const {
+      chatEnabled,
+      configData,
+      displayChat,
+      extraUserContent,
+      overallMaxViewerCount,
+      playerActive,
+      sessionMaxViewerCount,
+      streamOnline,
+      streamStatusMessage,
+      userAvatarImage,
+      username,
+      viewerCount,
+      websocket,
+      windowHeight,
+      windowWidth,
+    } = state;
+
+    const {
+      version: appVersion,
+      logo = {},
+      socialHandles = [],
+      name: streamerName,
+      summary,
+      tags = [],
+      title,
+    } = configData;
+    const { small: smallLogo = TEMP_IMAGE, large: largeLogo = TEMP_IMAGE } = logo;
+
+    const bgLogo = { backgroundImage: `url(${smallLogo})` };
+    const bgLogoLarge = { backgroundImage: `url(${largeLogo})` };
+
+    const tagList = !tags.length ?
+      null :
+      tags.map((tag, index) => html`
+        <li key="tag${index}" class="tag rounded-sm text-gray-100 bg-gray-700 text-xs uppercase mr-3 p-2 whitespace-no-wrap">${tag}</li>
+      `);
+
+    const socialIconsList =
+      !socialHandles.length ?
+      null :
+      socialHandles.map((item, index) => html`
+        <li key="social${index}">
+          <${SocialIcon} platform=${item.platform} url=${item.url} />
+        </li>
+      `);
+
+    const mainClass = playerActive ? 'online' : '';
+    const streamInfoClass = streamOnline ? 'online' : ''; // need?
+
+    const shortHeight = windowHeight <= HEIGHT_SHORT_WIDE;
+    const singleColMode = windowWidth <= WIDTH_SINGLE_COL && !shortHeight;
+    const extraAppClasses = classNames({
+      'chat': displayChat,
+      'no-chat': !displayChat,
+      'single-col': singleColMode,
+      'bg-gray-800': singleColMode && displayChat,
+      'short-wide': shortHeight,
+    })
+
+    return (
```

Comment:
This component is a little big for my taste, but TBH I don't know of any super-easy ways to decompose it, other than maybe moving the header out, which doesn't do a ton to remove the inherent complexity.

My guess is this will be a "hot" source of changes in the future, so maybe it's just worth keeping an eye on this and trying to tease out parts if it starts to become apparent?

Reply:
Yes I definitely had those same thoughts as well, such as breaking out the stream-info and user-info sections. Yeah I figure we'll keep an eye on things and see how they'll evolve. Hopefully not too much :D